### PR TITLE
Add formatter and smart indentation tests for checked operators

### PR DESCRIPTION
**Pull request by dibarbet**
also added a couple for user defined operators since those didn't exist either

### CODE REVIEW

**Review by davidwengier — APPROVED** [review]

File: src/EditorFeatures/CSharpTest/Formatting/FormattingEngineTests.cs [hunk 1]
```diff
@@ -2289,9 +2289,129 @@ class Test
...
+            AssertFormatWithView(expected, code, parseOptions: CSharpParseOptions.Default.WithLanguageVersion(LanguageVersion.Preview));
+        }
+
+        [WpfFact]
+        public void FormatUserDefinedCheckedOperatorOnType()
+        {
+            var code = @"
+interface I1
+{
+    abstract static I1 operator checked + ( I1 x, I1 y);$$
```
Comment (AlekseyTs) [inline src/EditorFeatures/CSharpTest/Formatting/FormattingEngineTests.cs:2400]: >abstract static I1 operator checked + ( I1 x, I1 y); [](http://example.com/codeflow?start=4&length=52)

Consider also testing unary and explicit conversion operators for completeness.

File: src/EditorFeatures/CSharpTest/Formatting/FormattingEngineTests.cs [hunk 2]
```diff
@@ -2289,9 +2289,129 @@ class Test
...
+            AssertFormatAfterTypeChar(code, expected);
+        }
+
+        [WpfFact]
+        public void FormatUserDefinedCheckedOperator()
+        {
+            var code = @"$$
+class C
+{
+    public static C operator checked + ( C x, C y){
```
Comment (AlekseyTs) [inline src/EditorFeatures/CSharpTest/Formatting/FormattingEngineTests.cs:2358]: >public static C operator checked + ( C x, C y){ [](http://example.com/codeflow?start=4&length=47)

Consider also testing unary operators for completeness.

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 2)

File: src/EditorFeatures/CSharpTest/Formatting/FormattingEngineTests.cs (hunk 1 above)
Comment (dibarbet) [reply src/EditorFeatures/CSharpTest/Formatting/FormattingEngineTests.cs:2478]: good call!  Found a bug with format on type for cast operators, see https://github.com/dotnet/roslyn/pull/60375/commits/6e3a5930aab38b22f8e1d5fe7e9a11cfdcaecfd2#diff-8dbfd863823b61ac2d84d6331f2b45d71ae69ea52d48b098ffb08b922abc7ce7R129

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Utilities/FormattingRangeHelper.cs [hunk 3]
```diff
@@ -125,7 +125,8 @@ MethodDeclarationSyntax or
                 PropertyDeclarationSyntax or
                 ConstructorDeclarationSyntax or
                 DestructorDeclarationSyntax or
-                OperatorDeclarationSyntax)
+                OperatorDeclarationSyntax or
+                ConversionOperatorDeclarationSyntax)
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Utilities/FormattingRangeHelper.cs:129]: aamzing.

**Review by CyrusNajmabadi — APPROVED** [review]

**Pull request merged**